File: java/micro/Sender.java
import java.io.OutputStream;
import java.net.Socket;

public class Sender {
    public static void main(String[] args) throws Exception {
        try (Socket socket = new Socket("127.0.0.1", 9000)) {
            OutputStream out = socket.getOutputStream();
            for (int i = 0; i < 5; i++) {
                byte cmd = 0x01;
                byte[] data = ("Hello " + i).getBytes();
                byte[] frame = MCPProtocol.packFrame(cmd, data);
                out.write(frame);
                System.out.printf("Sent: cmd=%d, data=%s\n", cmd, new String(data));
                Thread.sleep(1000);
            }
        }
    }
} 

File: mcp_micro/sender.py
import socket
import time
from mcp_protocol import MCPProtocol

HOST = '127.0.0.1'
PORT = 9000

if __name__ == '__main__':
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.connect((HOST, PORT))
        for i in range(5):
            cmd = 0x01  # 示例命令
            data = f'Hello {i}'.encode()
            frame = MCPProtocol.pack_frame(cmd, data)
            s.sendall(frame)
            print(f'Sent: cmd={cmd}, data={data}')
            time.sleep(1) 